### PR TITLE
[DOC] Add an example for _binned_statistic_dd

### PR DESCRIPTION
**Pull request by makbigc**
- follow up #10113

### CODE REVIEW

File: scipy/stats/_binned_statistic.py [hunk 1]
```diff
@@ -459,6 +459,44 @@ def binned_statistic_dd(sample, values, statistic='mean',
 
     .. versionadded:: 0.11.0
 
+    Examples
+    --------
+    >>> import numpy as np
```
Comment (tylerjereddy) [inline scipy/stats/_binned_statistic.py]: If I'm not mistaken, I think our documentation testing machinery will automatically provide `np`, so we don't need this.

**Comment by makbigc** [comment]
@tylerjereddy Anything else to implement? Please tell me.

File: scipy/stats/_binned_statistic.py [hunk 2]
```diff
@@ -459,6 +459,43 @@ def binned_statistic_dd(sample, values, statistic='mean',
...
+
+    Take 600 2-D arrays as an example. `binned_statistic_dd` can handle arrays
+    of higher dimension `D`. But a plot of dimension `D+1` is required.
+
+    >>> mu = np.array([0., 1.])
+    >>> Sigma = np.array([[1., -0.5],[-0.5, 1.5]])
+    >>> F = stats.multivariate_normal(mu, Sigma)
+    >>> data = F.rvs(size=600)
+
+    Create bins and count how many arrays falls in each bin:
```
Comment (tylerjereddy) [inline scipy/stats/_binned_statistic.py]: `falls` -> `fall`, I think

File: scipy/stats/_binned_statistic.py [hunk 3]
```diff
@@ -459,6 +459,43 @@ def binned_statistic_dd(sample, values, statistic='mean',
...
+
+    >>> N = 60
+    >>> X = np.linspace(-3, 3, N)
+    >>> Y = np.linspace(-3, 4, N)
+    >>> ret = stats.binned_statistic_dd(data, np.arange(600), bins=[X, Y], statistic='count')
+    >>> dz = ret.statistic
+
+    Set the volumn and the location of bars:
```
Comment (tylerjereddy) [inline scipy/stats/_binned_statistic.py]: Should `volumn` be `volume`?

**Review by tylerjereddy — COMMENTED** [review]
Made a few more minor suggestions. The new docstring example [looks quite nice in the CI render](https://13836-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.stats.binned_statistic_dd.html#scipy.stats.binned_statistic_dd).

Would be good if a resident stats expert takes a quick look (it is not really my domain).

File: scipy/stats/_binned_statistic.py [hunk 4]
```diff
@@ -459,6 +459,43 @@ def binned_statistic_dd(sample, values, statistic='mean',
...
+    --------
+    >>> from scipy import stats
+    >>> import matplotlib.pyplot as plt
+    >>> from mpl_toolkits.mplot3d import Axes3D
+
+    Take 600 2-D arrays as an example. `binned_statistic_dd` can handle arrays
+    of higher dimension `D`. But a plot of dimension `D+1` is required.
+
+    >>> mu = np.array([0., 1.])
+    >>> Sigma = np.array([[1., -0.5],[-0.5, 1.5]])
```
Comment (rgommers) [inline scipy/stats/_binned_statistic.py]: minor: could you use lowercase variable names? `N` is fine, but `sigma`, `x`, `y` would be nice. And `F` --> `multinormal` (or similar, a bit more descriptive).

File: scipy/stats/_binned_statistic.py [hunk 5]
```diff
@@ -459,6 +459,43 @@ def binned_statistic_dd(sample, values, statistic='mean',
 
     .. versionadded:: 0.11.0
 
+    Examples
+    --------
+    >>> from scipy import stats
+    >>> import matplotlib.pyplot as plt
+    >>> from mpl_toolkits.mplot3d import Axes3D
+
+    Take 600 2-D arrays as an example. `binned_statistic_dd` can handle arrays
```
Comment (rgommers) [inline scipy/stats/_binned_statistic.py]: This is phrased a bit unusually. The shape here is `(600, 2)`, which would be good to show at the end:
```
>>> data.shape
(600, 2)
```
And for phrasing maybe _"Take an array of 600 (x, y) coordinates as an example"_?

File: scipy/stats/_binned_statistic.py [hunk 6]
```diff
@@ -459,6 +459,43 @@ def binned_statistic_dd(sample, values, statistic='mean',
...
+    >>> F = stats.multivariate_normal(mu, Sigma)
+    >>> data = F.rvs(size=600)
+
+    Create bins and count how many arrays fall in each bin:
+
+    >>> N = 60
+    >>> X = np.linspace(-3, 3, N)
+    >>> Y = np.linspace(-3, 4, N)
+    >>> ret = stats.binned_statistic_dd(data, np.arange(600), bins=[X, Y], statistic='count')
+    >>> dz = ret.statistic
```
Comment (rgommers) [inline scipy/stats/_binned_statistic.py]: I'd call `dz` something like "bincounts" for clarity.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @makbigc, nice example. A few minor comments, if you can address those then I'll merge this PR.

**Comment by makbigc** [comment]
@rgommers  Amended. Please tell me if you have anything else to implement.

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
Merged, thanks @makbigc!  

> Please tell me if you have anything else to implement.

there's lots, happy to get more of your help:) for example, this issue is similar: https://github.com/scipy/scipy/issues/9947

**Pull request merged**